File: skills/optimize-python-code/assets/examples/reproduction/repro.py
def consume_twice(values):
    return list(values), list(values)


actual = consume_twice(iter([1, 2]))
expected = ([1, 2], [1, 2])
print(f"actual={actual!r}")
print(f"expected={expected!r}")
raise SystemExit(0 if actual != expected else 1)

File: skills/optimize-rust-code/assets/examples/reproduction/repro.rs
fn main() {
    let calls = std::cell::Cell::new(0);
    let values = [1, 2, 3];
    let lazy = values.iter().map(|value| {
        calls.set(calls.get() + 1);
        value
    });
    drop(lazy);
    println!("actual calls={}", calls.get());
    println!("expected eager calls=3");
    assert_eq!(calls.get(), 0);
}
